src/service/url-replacements-impl.js

```diff
@@ -607,10 +610,12 @@ export class UrlReplacements {
    * @param {!Object<string, *>=} opt_bindings
    * @param {!Object<string, *>=} opt_collectVars
    * @param {boolean=} opt_sync
+   * @param {!Object<string, boolean>=} opt_whiteList Optional white list of names
+   *     that can be substituted.
    * @return {!Promise<string>|string}
    * @private
    */
-  expand_(url, opt_bindings, opt_collectVars, opt_sync) {
+  expand_(url, opt_bindings, opt_collectVars, opt_sync, opt_whiteList) {
```

Comment:
This is probably ok as a first start, but I think the final API will be drastically different when we need to support the whole table of enabled/disabled/overriden vars. In the meantime, we could even simply call `string.replace('CLIENT_X', value)` to get us past this point and closer to correct implementation.